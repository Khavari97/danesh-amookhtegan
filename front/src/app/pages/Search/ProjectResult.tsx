import React, { Component } from 'react';
import Loading from "../../components/customs/Loading/Loading";
import { Grid } from "../../components/material-ui";
import ProjectNotFound from "../../components/customs/ProjectNotFound/ProjectNotFound";
import { apiClient } from "../../api/_api";
import ProjectCard from "../../components/customs/ProjectCard/ProjectCard";
import { IRootState } from "../../redux/RootReducer";
import { connect } from "react-redux";
import { Filter, searchPage, setSearchText } from "../../redux/Search/Actions";
import { ISearchState } from "../../redux/Search/ActionTypes";
import { withRouter } from "react-router-dom";
import SearchProjectFilter from "./SearchProjectFilter";
import BasicPagination from "../../components/customs/Pagination/BasicPagination";
import SearchInput from "../../components/customs/SearchInput/SearchInput";


const mapStateToProps = (store: IRootState) => {
    return store.search
}

const mapDispatchToProps = (dispatch: any) => {
    return {
        FilterProjects: (status: string, category: string, skill: string, text: string, countPages: number) => dispatch(Filter(status, category, skill, text, countPages)),
        searchPage: (page: number) => dispatch(searchPage(page)),
        setSearchText: (text: string) => dispatch(setSearchText(text)),
    }
}
interface IState {
    data: any;
    page: any;
    number: any;
    text: any;
}

interface IProps extends ISearchState {
    location: any,
    FilterProjects: any;
    searchPage: any;
    setSearchText: any
}

class ProjectResult extends Component<IProps, IState> {
    constructor(props: IProps) {
        super(props);
        this.state = {
            data: [],
            page: 1,
            number: 1,
            text: ""

        }
    }

    handleChange = (event: React.ChangeEvent<unknown>, value: number) => {
        this.props.searchPage(value)
    };


    render() {

        return (
            <>
                <SearchInput data_testid="search-input" text={(event: any) => {
                    if (event.target.value.length >= 3 || event.target.value.length === 0) {
                        this.props.setSearchText(event.target.value)
                    }
                }}
                    default={this.props.text}
                />

                <Grid>
                    <SearchProjectFilter />
                </Grid>

                {
                    this.props.loading ? <Loading /> :
                        <Grid className='animate__animated animate__fadeIn' container spacing={2}>
                            {this.props.projects?.length !== 0 ?
                                this.props.projects?.map((project: any, index: number) => {
                                    return <Grid data-testid="project-card" item xs={12} sm={6} md={4}
                                        key={index}><ProjectCard {...project} /></Grid>
                                }) : <ProjectNotFound />}
                        </Grid>
                }
                <BasicPagination siblingCount={0} boundaryCount={2} page={this.props.page} count={this.props.countPages} onChange={this.handleChange} />
            </>
        );
    }

    componentDidMount() {
        this.props.searchPage(1)
    }

    componentDidUpdate(prevProps: IProps) {
        if (this.props.text !== prevProps.text) {
            this.props.FilterProjects(this.props.project_status, this.props.project_category, this.props.project_skill, this.props.text, this.props.page)
        }
        if (this.props.page !== prevProps.page) {
            this.props.FilterProjects(this.props.project_status, this.props.project_category, this.props.project_skill, this.props.text, this.props.page)
        }
    }


}

export default withRouter(connect(mapStateToProps, mapDispatchToProps)(ProjectResult));
