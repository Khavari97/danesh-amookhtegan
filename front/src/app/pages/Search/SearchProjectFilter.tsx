import React, { Component } from 'react';
import { rootState } from "../../redux/RootReducer";
import { GetParentSkills, GetSkills, } from "../../redux/User/Profile/Actions";
import { connect } from "react-redux";
import { Grid } from "@material-ui/core";
import Autocomplete from "@material-ui/lab/Autocomplete";
import TextField from "@material-ui/core/TextField";
import { Filter, getAllSkills, setProjectCategory, setProjectSkill, setProjectStatus } from '../../redux/Search/Actions';

interface IState {
    fields: any,
    filter: { category: string, status: string, skill: string }
}

interface IProps {
    status: any;
    parentSkill: any;
    childSkill: any;
    GetSkills: any;
    GetParentSkills: any;
    FilterProjects: any;
    setProjectCategory: any;
    page: number;
    project_category: string;
    project_skill: string;
    project_status: string;
    setProjectStatus: any;
    text: string;
    getAllSkills: any;
    allSkills: [];
    setProjectSkill: any

}

const mapStateToProps = (state: rootState) => {
    return {
        status: state.project?.status,
        parentSkill: state.userProfile?.parentSkills,
        childSkill: state.userProfile?.skills,
        countPages: state.search?.countPages,
        page: state.search?.page,
        project_category: state.search?.project_category,
        project_skill: state.search?.project_skill,
        project_status: state.search?.project_status,
        text: state.search?.text,
        allSkills: state.search?.allSkills
    }
}

const mapDispatchToProps = (dispatch: any) => {
    return {
        GetSkills: () => dispatch(GetSkills()),
        GetParentSkills: () => dispatch(GetParentSkills()),
        FilterProjects: (status: string, category: string, skill: string, text: string, countPages: number) => dispatch(Filter(status, category, skill, text, countPages)),
        setProjectCategory: (category: string) => dispatch(setProjectCategory(category)),
        setProjectStatus: (status: string) => dispatch(setProjectStatus(status)),
        setProjectSkill: (skill: string) => dispatch(setProjectSkill(skill)),
        getAllSkills: () => dispatch(getAllSkills())
    }
}

class SearchProjectFilter extends Component<IProps, IState>{
    constructor(props: IProps) {
        super(props);
        this.state = {
            filter: { status: '', category: '', skill: '' },
            fields: [
                {
                    id: 'selectInput',
                    type: 'text',
                    name: ' status ',
                    label: ' وضعیت ',
                    value: ' ',
                    items: []
                },
                {
                    id: 'selectInput',
                    type: 'text',
                    name: ' parentSkill ',
                    label: ' دسته بندی ',
                    value: ' ',
                    items: props.parentSkill
                },
                {
                    id: 'selectInput',
                    type: 'text',
                    name: ' childSkill ',
                    label: ' مهارت ',
                    value: ' ',
                    items: props.childSkill
                }
            ]

        }
    }

    onChangeHandler = (event: any, values: any) => {
        console.log(values)
        this.props.setProjectCategory((values?.name) ? values.name : '')
    }

    render() {
        return (
            <Grid container>
                <Grid item data-test="sorting" style={{ padding: "0 5px 15px 0" }} sm={12}> دسته بندی بر اساس : </Grid>
                {
                    this.state.fields.map((field: any,index:number) => {
                        return (
                            <Grid key={index} item sm={4} xs={12}>
                                <Autocomplete
                                    id="combo-box-demo"
                                    data-test="Autocomplete"
                                    options={field.items}
                                    getOptionLabel={(option: any) => option.name}
                                    onChange={field.handleChange}
                                    size="small"
                                    style={{
                                        padding: "5px 5px",
                                        marginBottom: "20px"
                                    }}
                                    data-testid={field.name}
                                    renderInput={(params) =>
                                        <TextField {...params} label={field.label} variant="outlined" data-test="field" />}
                                />

                            </Grid>
                        )
                    })
                }

            </Grid>

        );
    }

    componentDidMount() {
        this.props.GetSkills();
        this.props.GetParentSkills();
        this.props.getAllSkills();
        this.props.FilterProjects('', '', '', this.props.text, this.props.page)
    }

    componentDidUpdate(prevProps: IProps) {
        if (this.props.project_category !== prevProps.project_category) {
            this.props.FilterProjects(this.props.project_status, this.props.project_category, this.props.project_skill, this.props.text, this.props.page)
        }
        if (this.props.project_status !== prevProps.project_status) {
            this.props.FilterProjects(this.props.project_status, this.props.project_category, this.props.project_skill, this.props.text, this.props.page)
        }
        if (this.props.project_skill !== prevProps.project_skill) {
            this.props.FilterProjects(this.props.project_status, this.props.project_category, this.props.project_skill, this.props.text, this.props.page)
        }
    }

    static getDerivedStateFromProps(nextProps: IProps, nextState: IState) {
        return {
            fields: [

                {
                    id: 'selectInput',
                    type: 'text',
                    name: ' status ',
                    label: ' وضعیت ',
                    value: ' ',
                    handleChange: (event: any, values: any) => {
                        nextProps.setProjectStatus((values?.code) ? values.code : '')
                    },
                    items: [
                        {
                            code: "STARTED",
                            name: "شروع شده"
                        },
                        {
                            code: "ENDED",
                            name: "پایان یافته"
                        },
                        {
                            code: "WAITING",
                            name: "در حال انتظار"
                        },
                        {
                            code: "DELETED",
                            name: "حذف شده"
                        }
                    ]
                },

                {
                    id: 'selectInput',
                    type: 'text',
                    name: ' parentSkill ',
                    label: ' دسته بندی ',
                    value: ' ',
                    handleChange: (event: any, values: any) => {
                        nextProps.setProjectCategory((values?.name) ? values.name : '')
                    },
                    items: nextProps.parentSkill
                },
                {
                    id: 'selectInput',
                    type: 'text',
                    name: ' childSkill ',
                    label: ' مهارت ',
                    value: ' ',
                    items: nextProps.allSkills,
                    handleChange: (event: any, values: any) => {
                        nextProps.setProjectSkill((values?.name) ? values.name : '')
                    },
                }
            ]

        }
    }
}
export default connect(mapStateToProps, mapDispatchToProps)(SearchProjectFilter)