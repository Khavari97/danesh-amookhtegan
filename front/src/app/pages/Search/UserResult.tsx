import React, { Component } from 'react';
import Loading from "../../components/customs/Loading/Loading";
import { Grid } from "../../components/material-ui";
import MemberCard from "../../components/customs/MemberCard/MemberCard";
import UserNotFound from "../../components/customs/UserNotFound/UserNotFound";
import { IRootState } from "../../redux/RootReducer";
import { FilterUsers, searchUsersPage, setSearchText } from "../../redux/Search/Actions";
import { connect } from "react-redux";
import { ISearchState } from "../../redux/Search/ActionTypes";
import { withRouter } from "react-router-dom";
import SearchUserFilter from "./SearchUserFilter";
import BasicPagination from "../../components/customs/Pagination/BasicPagination";
import SearchInput from "../../components/customs/SearchInput/SearchInput";


const mapStateToProps = (store: IRootState) => {
    return store.search
}

const mapDispatchToProps = (dispatch: any) => {
    return {
        searchUsersPage: (page: number) => dispatch(searchUsersPage(page)),
        setSearchText: (text: string) => dispatch(setSearchText(text)),
        FilterUsers: (university: string, skill: string, city: string, province: string, text: string, countPages: number) => dispatch(FilterUsers(university, skill, city, province, text, countPages)),
    }
}

interface IState {
    data: any;
    page: any;
    number: any;
    text: any;
    search: any
}

interface IProps extends ISearchState {
    location: any
    searchUsersPage: any
    FilterUsers: any
    setSearchText: any
}


class UserResult extends Component<IProps, IState> {
    constructor(props: IProps) {
        super(props);
        this.state = {
            data: [],
            page: 1,
            number: 1,
            text: "",
            search: []
        }
    }

    handleChange = (event: React.ChangeEvent<unknown>, value: number) => {
        this.props.searchUsersPage(value)
    };

    render() {
        return (
            <>
                <SearchInput data_testid='search-input' text={(event: any) => {
                    if (event.target.value.length >= 3 || event.target.value.length === 0) {
                        this.props.setSearchText(event.target.value)
                    }
                }}
                    default={this.props.text}
                />

                <Grid>
                    <SearchUserFilter />
                </Grid>
                {
                    this.props.loading ? <Loading /> :
                        <Grid className='animate__animated animate__fadeIn' container spacing={2}>
                            {this.props.users?.length !== 0 ?
                                this.props.users.map((user: any, index: number) => {
                                    return <Grid data-testid="card_user" item xs={12} sm={6} md={4} key={index}>
                                        <MemberCard {...user} /></Grid>
                                }) : <UserNotFound />}

                        </Grid>
                }
                <BasicPagination siblingCount={0} boundaryCount={2} page={this.props.pageUsers} count={this.props.countUserPages} onChange={this.handleChange} />
            </>
        );
    }
    componentDidMount() {
        this.props.searchUsersPage(1)
    }
    componentDidUpdate(prevProps: IProps) {
        if (this.props.pageUsers !== prevProps.pageUsers) {
            this.props.FilterUsers(this.props.user_university, this.props.user_skill, this.props.user_city, this.props.user_province, this.props.text, this.props.pageUsers)
        }
    }
}


export default withRouter(connect(mapStateToProps, mapDispatchToProps)(UserResult));