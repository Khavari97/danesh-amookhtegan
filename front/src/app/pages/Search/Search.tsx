import React, { Component } from 'react';
import { Container, Typography } from "@material-ui/core";
import CustomTabs, { TabBarProps } from "../../components/customs/CustomTabs/CustomTabs";
import ProjectResult from "./ProjectResult";
import UserResult from "./UserResult";
import SearchInput from "../../components/customs/SearchInput/SearchInput";


class Search extends Component<any> {

    SearchTabs: TabBarProps[] = [
        { label: "پروژه", component: <ProjectResult /> },
        { label: "افراد", component: <UserResult /> },
    ]

    render() {
        return (
            <Container>
                {this.props.homeVersion ? "" : <Typography data-testid="h1" variant={"h1"} component={"h1"}>جستجو</Typography>}
                <CustomTabs tabs={this.SearchTabs} />
            </Container>
        )
    }
}


export default Search;