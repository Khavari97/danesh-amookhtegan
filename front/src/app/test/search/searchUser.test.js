import React from 'react';
import Enzyme, { shallow} from "enzyme";
import Search from "../../pages/Search/SearchUserFilter";
import Autocomplete from "@material-ui/lab/Autocomplete";
import TextField from "@material-ui/core/TextField";
import EnzymeAdapter from "enzyme-adapter-react-16";
import store from '../../redux/store'
import { Provider } from 'react-redux'
import renderer from "react-test-renderer";
import sinon from "sinon";

Enzyme.configure({ adapter: new EnzymeAdapter() });

const setup = (props = {}) => {
  return shallow(
    <Provider store ={store}>
            <Search {...props} />
    </Provider>

  );
};

  const findByTestAttr = (wrapper, val) => {
    return wrapper.find(`[data-test="${val}"]`);
  };

describe('Test case for testing Autocomplete', () => {

    test('render searchUser component without error',()=>{
        let wrapper = setup();
        expect(wrapper).toBeTruthy();
        expect(wrapper.length).toEqual(1);
    })

    test ('handle header text :' , ()=>{
         const wrapper = setup().render();
         let field = findByTestAttr(wrapper, 'sorting')
         expect(field.text()).toBe(' دسته بندی بر اساس : ')
    })

    test('field input render check', () => {
        const wrapper = setup().render();
        let field = findByTestAttr(wrapper, 'Autocomplete')
        expect(field.length).toEqual(0);
    })

    test('should have value',()=>{
        const wrapper = setup();
        let field = findByTestAttr(wrapper, 'Autocomplete');
        wrapper.setProps({
            skills: "string",
            university:"kharazmi",
            province:"tehran",
            city:"tehran",
            page:1,
        })
        expect(field).toBeTruthy();
        expect(wrapper.length).toEqual(1);
    })

    it('check props not render without class',()=>{
        let wrapper = setup();

         let props = {
             label: 'button', onClick: () => { }
         }

          wrapper.setProps(props)

          expect(wrapper.props()).not.toEqual(props);
          expect(wrapper.props().label).toBeUndefined()

    })

    it('simulate button',()=>{
        let wrapper = setup();
        const mockCallBack = sinon.spy();
        let props = {
            label: 'button', onClick: mockCallBack
        }
        wrapper.setProps(props)
        wrapper.update();
        expect(mockCallBack).toHaveProperty('callCount', 0);
    })

    test('snapshot',()=>{
        const shot =(<Provider store={store}><Search/></Provider>);
        const snapshot = renderer.create(shot).toJSON();
        expect(snapshot).toMatchSnapshot();
    })
})

