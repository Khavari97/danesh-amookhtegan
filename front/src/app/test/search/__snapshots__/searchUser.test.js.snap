// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test case for testing Autocomplete snapshot 1`] = `
<div
  className="MuiGrid-root MuiGrid-container"
>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-sm-12"
    data-test="sorting"
    style={
      Object {
        "padding": "0 5px 15px 0",
        "width": "100%",
      }
    }
  >
     دسته بندی بر اساس : 
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
  >
    <div
      aria-expanded={false}
      aria-owns={null}
      className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
      data-testid="skills"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="combobox"
      style={
        Object {
          "marginBottom": "20px",
          "padding": "5px 3px",
        }
      }
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="combo-box-demo"
          id="combo-box-demo-label"
        >
           مهارت 
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          onClick={[Function]}
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-invalid={false}
            autoCapitalize="none"
            autoComplete="off"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            disabled={false}
            id="combo-box-demo"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            required={false}
            spellCheck="false"
            type="text"
            value=""
          />
          <div
            className="MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Clear"
              className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Clear"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Open"
              className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Open"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-3"
            >
              <span>
                 مهارت 
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
  >
    <div
      aria-expanded={false}
      aria-owns={null}
      className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
      data-testid="university"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="combobox"
      style={
        Object {
          "marginBottom": "20px",
          "padding": "5px 3px",
        }
      }
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="combo-box-demo"
          id="combo-box-demo-label"
        >
           دانشگاه 
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          onClick={[Function]}
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-invalid={false}
            autoCapitalize="none"
            autoComplete="off"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            disabled={false}
            id="combo-box-demo"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            required={false}
            spellCheck="false"
            type="text"
            value=""
          />
          <div
            className="MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Clear"
              className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Clear"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Open"
              className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Open"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-3"
            >
              <span>
                 دانشگاه 
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
  >
    <div
      aria-expanded={false}
      aria-owns={null}
      className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
      data-testid="province"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="combobox"
      style={
        Object {
          "marginBottom": "20px",
          "padding": "5px 3px",
        }
      }
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="combo-box-demo"
          id="combo-box-demo-label"
        >
           استان 
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          onClick={[Function]}
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-invalid={false}
            autoCapitalize="none"
            autoComplete="off"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            disabled={false}
            id="combo-box-demo"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            required={false}
            spellCheck="false"
            type="text"
            value=""
          />
          <div
            className="MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Clear"
              className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Clear"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Open"
              className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Open"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-3"
            >
              <span>
                 استان 
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
  <div
    className="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-12 MuiGrid-grid-sm-3"
  >
    <div
      aria-expanded={false}
      aria-owns={null}
      className="MuiAutocomplete-root MuiAutocomplete-hasClearIcon MuiAutocomplete-hasPopupIcon"
      data-testid="city"
      onClick={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="combobox"
      style={
        Object {
          "marginBottom": "20px",
          "padding": "5px 3px",
        }
      }
    >
      <div
        className="MuiFormControl-root MuiTextField-root MuiFormControl-fullWidth"
      >
        <label
          className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-marginDense MuiInputLabel-outlined"
          data-shrink={false}
          htmlFor="combo-box-demo"
          id="combo-box-demo-label"
        >
           شهر
        </label>
        <div
          className="MuiInputBase-root MuiOutlinedInput-root MuiAutocomplete-inputRoot MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiOutlinedInput-adornedEnd MuiInputBase-marginDense MuiOutlinedInput-marginDense"
          onClick={[Function]}
        >
          <input
            aria-activedescendant={null}
            aria-autocomplete="list"
            aria-controls={null}
            aria-invalid={false}
            autoCapitalize="none"
            autoComplete="off"
            autoFocus={false}
            className="MuiInputBase-input MuiOutlinedInput-input MuiAutocomplete-input MuiAutocomplete-inputFocused MuiInputBase-inputAdornedEnd MuiOutlinedInput-inputAdornedEnd MuiInputBase-inputMarginDense MuiOutlinedInput-inputMarginDense"
            disabled={false}
            id="combo-box-demo"
            onAnimationStart={[Function]}
            onBlur={[Function]}
            onChange={[Function]}
            onFocus={[Function]}
            onMouseDown={[Function]}
            required={false}
            spellCheck="false"
            type="text"
            value=""
          />
          <div
            className="MuiAutocomplete-endAdornment"
          >
            <button
              aria-label="Clear"
              className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-clearIndicator"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Clear"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root MuiSvgIcon-fontSizeSmall"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M19 6.41L17.59 5 12 10.59 6.41 5 5 6.41 10.59 12 5 17.59 6.41 19 12 13.41 17.59 19 19 17.59 13.41 12z"
                  />
                </svg>
              </span>
            </button>
            <button
              aria-label="Open"
              className="MuiButtonBase-root MuiIconButton-root MuiAutocomplete-popupIndicator"
              disabled={false}
              onBlur={[Function]}
              onClick={[Function]}
              onDragLeave={[Function]}
              onFocus={[Function]}
              onKeyDown={[Function]}
              onKeyUp={[Function]}
              onMouseDown={[Function]}
              onMouseLeave={[Function]}
              onMouseUp={[Function]}
              onTouchEnd={[Function]}
              onTouchMove={[Function]}
              onTouchStart={[Function]}
              tabIndex={-1}
              title="Open"
              type="button"
            >
              <span
                className="MuiIconButton-label"
              >
                <svg
                  aria-hidden={true}
                  className="MuiSvgIcon-root"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M7 10l5 5 5-5z"
                  />
                </svg>
              </span>
            </button>
          </div>
          <fieldset
            aria-hidden={true}
            className="PrivateNotchedOutline-root-1 MuiOutlinedInput-notchedOutline"
          >
            <legend
              className="PrivateNotchedOutline-legendLabelled-3"
            >
              <span>
                 شهر
              </span>
            </legend>
          </fieldset>
        </div>
      </div>
    </div>
  </div>
</div>
`;
